components/org.wso2.carbon.identity.authz.valve/src/main/java/org/wso2/carbon/identity/authz/valve/AuthorizationValve.java

```diff
@@ -238,4 +256,30 @@ private boolean isClientEmpty(AuthenticationContext authenticationContext) {
 
         return authenticationContext.getProperty(Constants.AUTH_CONTEXT_OAUTH_APP_PROPERTY) == null;
     }
+
+    /**
+     * Checks the endpoint is allowed to skip authorization for the given auth handler.
+     *
+     * @param authHandlerName   Name of the auth Handler
+     * @param requestUri    Request URI
+     *
+     * @return true if endpoint is allowed to skip authorization.
+     */
+    private boolean isAuthorizationSkipped(String authHandlerName, String requestUri) {
+
+        String[] authorizationSkipAllowedEndpointConfig = AuthConfigurationUtil.getInstance()
+                .getSkipAuthorizationAllowedEndpoints().get(authHandlerName);
+        if (authorizationSkipAllowedEndpointConfig == null) {
```

Comment:
check for empty as well

Reply:
If the `authorizationSkipAllowedEndpointConfig` will be empty, if empty value is configured for a auth handler. we should not allow to skip any endpoint such scenario. isn't?